frappe/email/doctype/notification/notification.js

```diff
@@ -59,9 +61,19 @@ frappe.notification = {
 			let receiver_fields = [];
 			if (frm.doc.channel === 'Email') {
 				receiver_fields = $.map(fields, function(d) {
-					return d.options == 'Email' ||
-						(d.options == 'User' && d.fieldtype == 'Link')
-						? get_select_options(d) : null;
+
+					if (d.fieldtype == 'Table') {
+						let child_fields = frappe.get_doc('DocType', d.options).fields;
+						return $.map(child_fields, function(df) {
+							return df.options == 'Email' ||
+								(df.options == 'User' && df.fieldtype == 'Link')
+								? get_select_options(df, d.fieldname) : null;
+						});
+					} else {
```

Comment:
what is this additional thing?

Reply:
This adds fields with options as "User" and "Email" from child tables in the select dropdown in the `recipients` table same way we do for fields on parent level

Reply:
ok can you add comments, because it is hard to read the logic.

also add comments in other places where you think it will be helpful in review ;-)